gpii/node_modules/processReporter/src/ProcessReporter.js

```diff
@@ -0,0 +1,224 @@
+/**
+ * GPII Process Reporter
+ *
+ * Copyright 2015 OCAD University
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+"use strict";
+
+var fluid = require("infusion"),
+    os = require("os"),
+    gpii = fluid.registerNamespace("gpii");
+
+fluid.require("kettle", require);
+fluid.require("./processesBridge.js", require);
+
+fluid.defaults("gpii.processReporter", {
+    gradeNames: "fluid.component"
+});
+
+fluid.defaults("gpii.processReporter.base", {
+    gradeNames: ["kettle.app", "gpii.processReporter"],
+    requestHandlers: {
+        processGet: {
+            route: "/processes",
+            type: "gpii.processReporter.handlers.get",
+            method: "get"
+        },
+        processGetSingle: {
+            route: "/processes/:solutionID",
+            type: "gpii.processReporter.solution.get",
+            method: "get"
+        }
+    },
+    components: {
+        platformReporter: {
+            type: "gpii.platformReporter.native"
+        },
+        nameResolver: {
+            type: "gpii.processReporter.nameResolver"
+        }
+    },
+    invokers: {
+        fireResponse: {
+            funcName: "gpii.processReporter.fireResponse",
+            args: ["{arguments}.0", "{arguments}.1", "{platformReporter}"]
+        }
+    }
+});
+
+fluid.defaults("gpii.processReporter.handlers.get", {
+    gradeNames: ["kettle.request.http"],
+    invokers: {
+        handleRequest: {
+            func: "{processReporter}.get",
+            args: [
+                "{request}"
+            ]
+        }
+    }
+});
+
+fluid.defaults("gpii.processReporter.solution.get", {
+    gradeNames: ["kettle.request.http"],
+    invokers: {
+        handleRequest: {
+            func: "{processReporter}.getProcess",
+            args: ["{request}", "{processReporter}", "{solutionsRegistryDataSource}"]
+        }
+    }
+});
+
+fluid.defaults("gpii.processReporter.nameResolver", {
+    gradeNames: ["fluid.component"],
+    invokers: {
+        resolveName: {
+            funcName: "fluid.identity"
+        }
+    }
+});
+
+gpii.processReporter.fireResponse = function (request, installedSolutions, platformReporter) {
```

Comment:
Inconsistency in naming here - the callers of this function name this "runningSolutions"